src/persist/src/indexed/columnar.rs

```diff
@@ -396,6 +411,21 @@ impl ColumnarRecordsBuilder {
         debug_assert_eq!(self.borrow().validate(), Ok(()));
     }
 
+    /// Returns if the given key_offsets+key_data or val_offsets+val_data fits
+    /// in the limits imposed by ColumnarRecords.
+    ///
+    /// Note that limit is always [KEY_VAL_DATA_MAX_LEN] in production. It's
+    /// only override-able here for testing.
+    pub fn can_fit(&self, key: &[u8], val: &[u8], limit: usize) -> bool {
+        let key_data_size = (self.key_offsets.len() + 1) * BYTES_PER_KEY_VAL_OFFSET
```

Comment:
I'm going to just say the justification for this expression in a sentence to make sure I understand it correctly and you can tell me if its right

we'll be storing the offsets inline with the key data and in the offsets array. It's `(key_offsets.len() + 1) because we'll need to add a new offset for this new key we're inserting, and its multiplied by `BYTES_PER_KEY_VAL_OFFSET` because `key_offsets` is an array of `i32`.

those bytes are added with the existing key and existing value bytes and the whole thing has to be leq the limit.

Reply:
so are we storing two copies of the key and val offsets?